tests/foreman/api/test_config_template.py

```diff
@@ -30,3 +30,57 @@ def test_build_pxe_default(self):
         )
         response.raise_for_status()
         self.assertIsInstance(response.json(), dict)
+
+    def test_associate_config_template_with_organization(self):
+        """@Test: Associate config template with organization
+
+        @Assert: Config template is associated with organization
+
+        @Feature: ConfigTemplate
+
+        """
+        org_1_id = entities.Organization().create_json()['id']
+        org_2_id = entities.Organization().create_json()['id']
+
+        # By default, a configuration template should have no organizations.
+        conf_templ = entities.ConfigTemplate().create_json()
+        self.assertEqual(0, len(conf_templ['organizations']))
+
+        # Associate our configuration template with one organization.
+        client.put(
+            entities.ConfigTemplate(id=conf_templ['id']).path(),
+            {'config_template': {'organization_ids': [org_1_id]}},
+            verify=False,
+            auth=get_server_credentials(),
+        ).raise_for_status()
+
+        # Verify that the association succeeded.
+        new_attrs = entities.ConfigTemplate(id=conf_templ['id']).read_json()
+        self.assertEqual(1, len(new_attrs['organizations']))
+        self.assertEqual(org_1_id, new_attrs['organizations'][0]['id'])
```

Comment:
This can be simplified by working with `ConfigTemplate` objects instead of dicts of data.

```python
conf_templ = conf_templ.read()
self.assertEqual(1, len(conf_templ.organizations))
self.assertEqual(org_1_id, conf_templ.organizations[0].id)
```

When you call `read`, a brand new object is created, populated and returned. That's why you have to write `conf_templ = conf_templ.read()`. If you failed to do the assignment, then a new `ConfigTemplate` object would be created, populated, returned - and immediately discarded.